### PR TITLE
[libclipboard] new port 

### PR DESCRIPTION
**Pull request by judemille**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by judemille** [comment]
This repository is licensed under the MIT license. What reason is there to completely reassign my copyright to Microsoft? By contributing, I am licensing my changes under the MIT license. Is that not enough?

**Comment by judemille** [comment]
I can squash my crazy number of commits if need be.

**Comment by MonicaLiu0311** [comment]
> This repository is licensed under the MIT license. What reason is there to completely reassign my copyright to Microsoft? By contributing, I am licensing my changes under the MIT license. Is that not enough?

CC @data-queue  @BillyONeal

File: ports/libclipboard/fix-libclipboard-cmake.patch [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+ endif()
+ 
+ if (NOT (LIBCLIPBOARD_BUILD_WIN32 OR LIBCLIPBOARD_BUILD_X11 OR LIBCLIPBOARD_BUILD_COCOA))
+@@ -57,24 +58,6 @@ if (CMAKE_COMPILER_IS_GNUCXX OR LIBCLIPBOARD_BUILD_COCOA)
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -pedantic -g")
+ endif()
+ 
+-# Dependencies
+-if (LIBCLIPBOARD_BUILD_COCOA)
+-    set_source_files_properties(src/clipboard_cocoa.c PROPERTIES COMPILE_FLAGS "-x objective-c")
```
Comment (dg0yt) [inline ports/libclipboard/fix-libclipboard-cmake.patch:27]: Maybe the modern way is to set the `LANGUAGE` property to `OBJC` (as enabled as a language in this patch)?

File: ports/libclipboard/fix-libclipboard-cmake.patch [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+\ No newline at end of file
+diff --git a/cmake/unofficial-libclipboard-config.cmake b/cmake/unofficial-libclipboard-config.cmake
+new file mode 100644
+index 0000000..9d48704
+--- /dev/null
++++ b/cmake/unofficial-libclipboard-config.cmake
+@@ -0,0 +0,7 @@
++include(CMakeFindDependencyMacro)
++find_dependency(PkgConfig REQUIRED)
++if(UNIX AND NOT APPLE)
```
Comment (dg0yt) [inline ports/libclipboard/fix-libclipboard-cmake.patch:184]: I would prefer a configured
```suggestion
+if("@ LIBCLIPBOARD_BUILD_X11@")
```
here.

**Review by dg0yt — COMMENTED** [review]
The patch seem more invasive than necessary.

File: ports/libclipboard/fix-libclipboard-cmake.patch (hunk 1 above)
Comment (judemille) [reply ports/libclipboard/fix-libclipboard-cmake.patch:27]: I could try that. Frankly, I have no idea why the original author made an ObjC file with the extension `.c`.

File: ports/libclipboard/fix-libclipboard-cmake.patch (hunk 2 above)
Comment (judemille) [reply ports/libclipboard/fix-libclipboard-cmake.patch:184]: Good catch.

**Comment by judemille** [comment]
@microsoft-github-policy-service agree

I morally disagree with the CLA, but it's clear to me that unless I give Microsoft the right to use my work without so much as a mention of my existence, no movement will happen.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
The package libclipboard provides CMake targets:

    find_package(unofficial-libclipboard CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::libclipboard::clipboard)
```

File: ports/libclipboard/fix-libclipboard-cmake.patch [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
++    set_source_files_properties(src/clipboard_cocoa.c PROPERTIES LANGUAGE OBJC)
+-    set(LIBCLIPBOARD_PRIVATE_LIBS ${LIBCLIPBOARD_PRIVATE_LIBS} "-framework Cocoa")
+-elseif(LIBCLIPBOARD_BUILD_X11)
+-    include(FindPkgConfig REQUIRED)
+-    pkg_check_modules(X11 xcb REQUIRED)
+-    find_package(Threads REQUIRED)
+-
+-    include_directories(${X11_INCLUDE_DIRS})
+-    link_directories(${X11_LIBRARY_DIRS})
+-    set(LIBCLIPBOARD_PRIVATE_LIBS ${LIBCLIPBOARD_PRIVATE_LIBS} ${X11_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```
Comment (dg0yt) [inline ports/libclipboard/fix-libclipboard-cmake.patch:37]: Why is this removed?

File: ports/libclipboard/fix-libclipboard-cmake.patch [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+-include_directories("${CMAKE_CURRENT_BINARY_DIR}/include")
+-
+ # Configure header
+ configure_file("include/libclipboard-config.h.in" "include/libclipboard-config.h")
+ 
+@@ -87,66 +70,79 @@ set(HEADERS
+ set(SOURCE
+     src/clipboard_win32.c
+     src/clipboard_x11.c
+-    src/clipboard_cocoa.c
```
Comment (dg0yt) [inline ports/libclipboard/fix-libclipboard-cmake.patch:51]:
```suggestion
```

File: ports/libclipboard/fix-libclipboard-cmake.patch [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+-
+-# Make the library
+-add_library(clipboard ${HEADERS} ${SOURCE})
+-target_link_libraries(clipboard LINK_PRIVATE ${LIBCLIPBOARD_PRIVATE_LIBS})
+-if (LIBCLIPBOARD_ADD_SOVERSION)
+-    # Not by default because my file system doesn't support symlinks
+-    set_target_properties(clipboard PROPERTIES SOVERSION ${LIBCLIPBOARD_VERSION} VERSION ${LIBCLIPBOARD_VERSION})
++if(LIBCLIPBOARD_BUILD_COCOA)
++list(APPEND SOURCE src/clipboard_cocoa.c)
+ endif()
```
Comment (dg0yt) [inline ports/libclipboard/fix-libclipboard-cmake.patch:68]:
```suggestion
 endif()
```

File: ports/libclipboard/fix-libclipboard-cmake.patch [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+ 
+-# Testing mode?
+-if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/googletest/googletest)
```
Comment (dg0yt) [inline ports/libclipboard/fix-libclipboard-cmake.patch:71]: It is easy to ensure that the file doesn't exist, so we don't need to remove all these lines.

**Review by dg0yt — COMMENTED** [review]
I still think the patch is much too invasive, not minimal (as required by guidelines).
I also dislike the removal of the official pkg-config files.
An incomplete set of hints follows.
Disclaimer: I'm only a community member,

File: ports/libclipboard/fix-libclipboard-cmake.patch (hunk 4 above)
Comment (judemille) [reply ports/libclipboard/fix-libclipboard-cmake.patch:51]: This file will not build in the slightest (even with definition guards) without an Objective-C compiler, which we don't want to assume the presence of unless Cocoa is requested (read: on a Mac).

File: ports/libclipboard/fix-libclipboard-cmake.patch (hunk 5 above)
Comment (judemille) [reply ports/libclipboard/fix-libclipboard-cmake.patch:68]: See [my other comment.](https://github.com/microsoft/vcpkg/pull/34764/files/6fae7822fa7fda07f57b0a1bd062e3ac9964c70e#r1384370256)

File: ports/libclipboard/fix-libclipboard-cmake.patch (hunk 6 above)
Comment (judemille) [reply ports/libclipboard/fix-libclipboard-cmake.patch:71]: Fair enough.

File: ports/libclipboard/fix-libclipboard-cmake.patch (hunk 3 above)
Comment (judemille) [reply ports/libclipboard/fix-libclipboard-cmake.patch:37]: I'll admit, my brain went "not modern CMake, kill!". Perhaps I'll fork the repository to bring the CMake stuff up to standard, since it's not been updated in three years anyway.

**Comment by JavierMatosD** [comment]
I agree with @dg0yt regarding the patch. I like the modernization, but we should really try to follow upstream's lead whenever possible and keep patch sizes down to a minimum. I'll place the PR on draft for now.

**Comment by MonicaLiu0311** [comment]
If you're ready, please revert to "ready for review".

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.
